geopandas/io/tests/test_file.py

```diff
@@ -276,10 +278,12 @@ def test_read_file_datetime_invalid(tmpdir, ext, engine):
 
 @pytest.mark.parametrize("ext", dt_exts)
 def test_read_file_datetime_out_of_bounds_ns(tmpdir, ext, engine):
+    if engine == "pyogrio" and not (PANDAS_GE_20 and PYOGRIO_GE_07):
+        pytest.skip("requires pyogrio >= 0.7.0 and pandas >= 2.0 to pass")
```

Comment:
This is skipped because:

Older pandas (1.4.4, 1.5.3) with older pyogrio (0.5.0, 0.5.1) raise `pandas._libs.tslibs.np_datetime.OutOfBoundsDatetime: Out of bounds nanosecond timestamp: 9999-12-31 00:00:00`

New pandas (2.2.1) in combination with old pyogrio (0.5.3) raise `AssertionError: assert dtype('<M8[ms]') == 'object'`.

And pandas 1.5.3 with recent pyogrio lead to `UnboundLocalError: local variable 'res' referenced before assignment` which has been fixed on pyogrio main in https://github.com/geopandas/pyogrio/pull/370

Reply:
I suppose technically for the 
> New pandas (2.2.1) in combination with old pyogrio (0.5.3) raise AssertionError: assert dtype('<M8[ms]') == 'object'.

case we could update the expectation to be for a ms resolution datetime. From what you're saying it seems that new pandas and new pyogrio passes the test as is? That's perhaps an oversight, there I would think it should give a lower resolution datetime to accomodate (especially since GDAL doesn't suport nanosecond precision anyway)

Reply:
> we could update the expectation to be for a ms resolution datetime 

not sure if it is worth the effort of figuring out which combinations of versions require this.

> From what you're saying it seems that new pandas and new pyogrio passes the test as is? That's perhaps an oversight, there I would think it should give a lower resolution datetime to accomodate (especially since GDAL doesn't suport nanosecond precision anyway)

Yes, latest pandas with latest pyogrio pass fine. Which I believe is according to the expectations of this test, which checks that the out of bounds datetime is not a datetime object but a string and the resulting array dtype is object.